src/preact/base-element.js

```diff
@@ -523,23 +520,32 @@ export class PreactBaseElement extends AMP.BaseElement {
   }
 
   /** @private */
-  onLoad_() {
-    if (this.loadDeferred_) {
-      this.loadDeferred_.resolve();
-      this.loadDeferred_ = null;
-      dispatchCustomEvent(this.element, 'load', null, {bubbles: false});
+  maybeUpdateReadyState_() {
+    const {currentRef_: api} = this;
+
+    const apiReadyState = api && api['readyState'];
+    if (apiReadyState && apiReadyState !== this.element.readyState) {
+      this.onReadyState_(apiReadyState);
     }
   }
 
   /**
-   * @param {*} opt_reason
+   * @param {!ReadyState} state
+   * @param {*=} opt_failure
    * @private
    */
-  onLoadError_(opt_reason) {
-    if (this.loadDeferred_) {
-      this.loadDeferred_.reject(opt_reason || new Error('load error'));
-      this.loadDeferred_ = null;
-      dispatchCustomEvent(this.element, 'error', null, {bubbles: false});
+  onReadyState_(state, opt_failure) {
+    this.setReadyState(state, opt_failure);
+
+    const Ctor = this.constructor;
+    if (Ctor['unloadOnPause']) {
+      this.updateIsPlaying_(state == ReadyState.COMPLETE);
+    }
```

Comment:
I don't understand this. What does `unloadOnPause` have to do with the ready state? Did you mean `loadable`?

Reply:
There are two cases here: 

1. `!onloadOnPause`: a component knows when it's playing exactly and can pause itself. A simplest one - `<video>` tag. In this case we only need track its "displayability" while the component is playing.
2. `unloadOnPause`: an iframe embed. Once it's loaded - we unfortunately not sure if/when it's playing and how to pause it. What this code does: as soon as the component has loaded (`readyState == COMPLETE`) I consider it's playing (or rather "could be playing"). This in turn monitor's the components displayability to pause it when it's no longer displayable (no longer has size).

In other words, this code basically says "we believe a component might be playing at any time once it's loaded".

Reply:
Please add that as a comment, that's a good explanation.